frontend/src/component/pages/shared/Footer.tsx

```diff
@@ -1,14 +1,57 @@
 import React, { Component } from 'react';
 
 import './Footer.scss';
+import { Update } from '../../../entity/Update';
+import { UIFactory } from '../../UIFactory';
+
+interface State {
+  showChangeLogModal: boolean;
+}
 
 interface Props {
+  uiFactory: UIFactory;
   authorName: string;
   authorPortfolio: string;
   version: string;
+  changeLog?: Array<Update>;
+  newUpdateReleased?: boolean;
+  updateLastSeenChangeLog?: () => void;
 }
 
-export class Footer extends Component<Props> {
+export class Footer extends Component<Props, State> {
+  state = {
+    showChangeLogModal: false
+  };
+
+  componentDidMount() {
+    this.setState({
+      showChangeLogModal: this.props.newUpdateReleased || false
+    });
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.newUpdateReleased !== this.props.newUpdateReleased) {
+      this.setState({
+        showChangeLogModal: this.props.newUpdateReleased || false
+      });
+    }
+  }
+
+  handleShowChangeLog = () => {
+    this.setState({
+      showChangeLogModal: true
+    });
```

Comment:
Let's propagate this event out of footer and delegate the handling logic to Home component so that we have more flexibilities of add more features.

The only job of view component to perform rendering. This allows us to reuse those component at somewhere else without breaking the existing code.